picard/ui/matchdetailsdialog/__init__.py

```diff
@@ -0,0 +1,278 @@
+        self.input_filename_lbl.setTextInteractionFlags(QtCore.Qt.TextSelectableByMouse)
+        self.verticalLayout.addWidget(self.input_filename_lbl)
+
+        self.tip_widget = QtWidgets.QWidget(self)
+        self.tip_lbl = QtWidgets.QLabel(self.tip_widget)
+        self.tip_lbl.setText(_("Tip: Column header left click to sort, right click to filter"))
+        self.tip_lbl.setMargin(10)
+        self.verticalLayout.addWidget(self.tip_lbl)
+
+        self.center_widget = QtWidgets.QWidget(self)
+        self.center_widget.setObjectName("center_widget")
+        self.center_layout = QtWidgets.QVBoxLayout(self.center_widget)
+        self.center_layout.setObjectName("center_layout")
+        self.center_layout.setContentsMargins(1, 1, 1, 1)
+        self.center_widget.setLayout(self.center_layout)
+        self.verticalLayout.addWidget(self.center_widget)
+
+        self.acoustid_lookup_url_widget = QtWidgets.QWidget(self)
+        self.acoustid_lookup_url = QtWidgets.QLabel(self.acoustid_lookup_url_widget)
+        self.acoustid_lookup_url.setOpenExternalLinks(True)
+        self.verticalLayout.addWidget(self.acoustid_lookup_url)
+
+        self.buttonBox = QtWidgets.QDialogButtonBox(self)
+        self.export_button = QtWidgets.QPushButton("Export Results", self.buttonBox)
+        self.export_button.setEnabled(True)
+        self.buttonBox.addButton(self.export_button, QtWidgets.QDialogButtonBox.ActionRole)
+        self.export_button.clicked.connect(self.export_data)
+
+        self.accept_button = QtWidgets.QPushButton("Load into Picard", self.buttonBox)
+        self.accept_button.setEnabled(False)
+        self.buttonBox.addButton(self.accept_button, QtWidgets.QDialogButtonBox.AcceptRole)
+
+        self.buttonBox.addButton(StandardButton(StandardButton.CANCEL), QtWidgets.QDialogButtonBox.RejectRole)
+        self.buttonBox.accepted.connect(self.accept)
+        self.buttonBox.rejected.connect(self.reject)
+        self.verticalLayout.addWidget(self.buttonBox)
+
+    # Override
+    def show_table(self, selected_row_nbr=-1, sort_column=None, sort_order=QtCore.Qt.DescendingOrder):
+        self.add_widget_to_center_layout(self.table)
+        self.table.selectRow(selected_row_nbr)
+        self.table.setCurrentCell(selected_row_nbr, 0)
+        self.table.horizontalHeader().setSortIndicatorShown(self.sorting_enabled)
+        self.table.horizontalHeader().setDefaultAlignment(QtCore.Qt.AlignBottom | QtCore.Qt.AlignCenter | QtCore.Qt.TextWordWrap)
+
+        # sort
+        self.table.setSortingEnabled(self.sorting_enabled)
+        if self.sorting_enabled and sort_column:
+            self.table.sortItems(self.colpos(sort_column), sort_order)
+
+        # Hide debug cols. using first char of label to mark it private
+        should_hide_debug_cols = not config.setting['show_debug_col_in_match_dialog']
+        for i in range(self.table.columnCount()):
+            is_debug_col = self.table.horizontalHeaderItem(i).text().startswith('_')
```

Comment:
Just a plain No to this.

Remove '_' from labels, add `set()` for shown (or hidden) columns.

```patch
diff --git a/picard/ui/matchdetailsdialog/__init__.py b/picard/ui/matchdetailsdialog/__init__.py
index 62f473f9..30e11e6b 100644
--- a/picard/ui/matchdetailsdialog/__init__.py
+++ b/picard/ui/matchdetailsdialog/__init__.py
@@ -260,7 +260,7 @@ class MatchDetailsDialogBase(TableBasedDialog):
         # Hide debug cols. using first char of label to mark it private
         should_hide_debug_cols = not config.setting['show_debug_col_in_match_dialog']
         for i in range(self.table.columnCount()):
-            is_debug_col = self.table.horizontalHeaderItem(i).text().startswith('_')
+            is_debug_col = self.colname(i) in self.debug_only_columns
             hide_this_col = is_debug_col and should_hide_debug_cols
             self.table.setColumnHidden(i, hide_this_col)
 
diff --git a/picard/ui/matchdetailsdialog/matchdetails.py b/picard/ui/matchdetailsdialog/matchdetails.py
index c23f3558..b51e6617 100644
--- a/picard/ui/matchdetailsdialog/matchdetails.py
+++ b/picard/ui/matchdetailsdialog/matchdetails.py
@@ -56,34 +56,46 @@ class MatchDetailsDialog(MatchDetailsDialogBase):
         self.file_ = None
         self.setWindowTitle(_("Match Details"))
         self.columns = [
-            ('nbr',                  "_Log\nSeq"),
-            ('match_dtl_entry_type', "_Entry Type"),
-            ('is_new',               "_Better?"),
-            ('rec_id',               "_Recording ID"),
+            ('nbr',                  "Log\nSeq"),
+            ('match_dtl_entry_type', "Entry Type"),
+            ('is_new',               "Better?"),
+            ('rec_id',               "Recording ID"),
             ('rec_title',            "Recording Title"),
             ('rec_len',              "Recording\nLength"),
             ('rec_artist',           "Recording\nArtist"),
-            ('rg_id',                "_Release Group ID"),
+            ('rg_id',                "Release Group ID"),
             ('rg_title',             "Release Group\nTitle"),
             ('rg_pri_type',          "Release Group\nPrimary Type"),
             ('rg_sec_type',          "Release Group\nSecondary Types"),
-            ('rel_id',               "_Release ID"),
+            ('rel_id',               "Release ID"),
             ('rel_ctry',             "Release\nCountry"),
             ('rel_dt',               "Release\nDate"),
             ('med_fmt',              "Medium\nFormat"),
             ('med_pos',              "Medium\nPosition"),
             ('med_tot_cnt',          "Medium\nCount"),
-            ('trk_id',               "_Track ID"),
+            ('trk_id',               "Track ID"),
             ('trk_pos',              "Track\nPosition"),
             ('med_trk_cnt',          "Medium\nTrack\nCount"),
             ('rel_tot_trk_cnt',      "Release\nTotal Track\nCount"),
-            ('parts',                "_Scoring Parts"),
-            ('release_parts',        "_Scoring Release Parts"),
-            ('recording_base_score', "_Recording\nBase Score\n(Sources Ratio)"),
+            ('parts',                "Scoring Parts"),
+            ('release_parts',        "Scoring Release Parts"),
+            ('recording_base_score', "Recording\nBase Score\n(Sources Ratio)"),
             ('acoust_id',            "AcoustId ID"),
             ('rec_src_cnt',          "AcoustId\nSource\nCount"),
             ('similarity',           "Similarity")
         ]
+        self.debug_only_columns = {
+            'is_new',
+            'match_dtl_entry_type',
+            'nbr',
+            'parts',
+            'rec_id',
+            'recording_base_score',
+            'release_parts',
+            'rel_id',
+            'rg_id',
+            'trk_id',
+        }
 
     def load_match_details(self, track, file_):
         """ Retrieves data and loads it to the screen/dialog box """
diff --git a/picard/ui/tablebaseddialog.py b/picard/ui/tablebaseddialog.py
index 3381f854..da35d2bd 100644
--- a/picard/ui/tablebaseddialog.py
+++ b/picard/ui/tablebaseddialog.py
@@ -115,6 +115,9 @@ class TableBasedDialog(PicardDialog):
     def colpos(self, colname):
         return self.__colkeys.index(colname)
 
+    def colname(self, pos):
+        return self.__colkeys[pos]
+
     @abstractmethod
     def get_value_for_row_id(self, row, value):
         pass
```